cnxarchive/views.py

```diff
@@ -115,6 +115,32 @@ def is_latest(id, version):
     return get_latest_version(id) == version
 
 
+def get_state(cursor, id, version):
+    """Determine the state of the content."""
+    version_split = version.split(".")
+    args = [id]
+    args.extend(version_split)
+
+    sql_statement = """
+    SELECT ms.statename
+    FROM modules as m
+    JOIN modulestates as ms
+    ON m.stateid=ms.stateid
+    WHERE uuid=%s
+    AND major_version=%s
+    """
+    if len(version_split) == 2:
+        sql_statement += "AND minor_version=%s"
+    sql_statement += ";"
+
+    cursor.execute(sql_statement, args)
```

Comment:
This can be written "WHERE ident_hash(uuid, major_version, minor_version) = %s" and "args='@'.join((id,version))", and skip the splitting of the version.